ppb/systems.py

```diff
@@ -40,7 +57,61 @@ def on_update(self, update, signal):
         for pygame_event in pygame.event.get():
             ppb_event = self.event_map.get(pygame_event.type)
             if ppb_event is not None:
-                signal(ppb_event())
+                signal(getattr(self, ppb_event)(pygame_event, update.scene))
```

Comment:
If you extract the `getattr()` call you can catch any `AttributeError` and give a more concise error if an event doesn't have a matching method.

Reply:
Big catch is that if the event exists in the lookup dictionary, it should exist on the system and failure here is definitely a thing to catch before it's in a user's hands.